docs/api/ref/AppInstance.md

```diff
@@ -0,0 +1,177 @@
+---
+id: AppInstance
+sidebar_label: AppInstance
+title: AppInstance
+hide_title: true
+---
+# `AppInstance`
+
+```ts
+interface AppInstance {
+  //properties
+  readonly instanceId : string;
+  readonly status : 'ready' | 'loading' | 'unregistered';
+  
+  //methods
+  addContextListener(handler: ContextHandler): Listener;
+  addContextListener(contextType: string, handler: ContextHandler): Listener;
```

Comment:
I still think we should fix the arg order for all the `addContextListener` methods 😁

Reply:
2.0?

Reply:
They actually cause a lot of problems, being out of order. The code you have to write to deal with it is really ugly.

Reply:
@rikoe @mattjamieson - if we made this change in 1.2, what would be the impact to existing apps?  Can some of the platform owners weigh in?  @kriswest @lspiro-Tick42 @tim-dinsdale

Reply:
Speaking of jumping through hoops to make function signatures work, the API would be easier to implement and more consistent if addContextListener() were to return a `Promise<Listener>` rather than a `Listener`.